app/src/main/java/ch/epfl/sdp/social/ChatRepository.java

```diff
@@ -174,34 +174,54 @@ protected void onPostExecute(List<Message> ls)
 
 
 
-    public static void insertMessageFromRemote(Timestamp tm, String content) {
+    public static void insertMessageFromRemote(Timestamp tm, String content, int chat_id) {
         new AsyncTask<Void, Void, Void>() {
             @Override
             protected Void doInBackground(Void... voids) {
-                //try {
-                    singleton.chatDB.daoAccess().sendMessage(new Message(tm.toDate(), content));
-                //}catch (SQLiteConstraintException e)
-                //{ }
+                try {
+                    singleton.chatDB.daoAccess().sendMessage(new Message(tm.toDate(), content, chat_id));
+                }catch (SQLiteConstraintException e)
+                { }
                 return null;
             }
         }.execute();
 
     }
 
-    public static void getChat(String current, String other)
+    public static Chat getChat(String current, String other)
     {
-        new AsyncTask<Void, Void, List<Chat>>() {
-
-            @Override
-            protected List<Chat> doInBackground(Void... voids) {
-                return singleton.chatDB.daoAccess().getChatFromCurrentToOther(current, other);
-            }
+        Chat output = null;
+        try {
+            output = new AsyncTask<Void, Void, List<Chat>>() {
+
+                @Override
+                protected List<Chat> doInBackground(Void... voids) {
+                    return singleton.chatDB.daoAccess().getChatFromCurrentToOther(current, other);
+                }
+
+                /*@Override
+                protected void onPostExecute(List<Chat> exists) {
+                    ((WaitOnChatRetrieval)(singleton.contextActivity)).chatFetched(exists);
+                }*/
+            }.execute().get().get(0);
```

Comment:
The whole point is **_asynchronous_** execution!!!. This task should not block just to get the result of the method "doInBackground." Sacha, please contact me regarding this.

Reply:
I know, but you cannot have it asynchronous. Otherwise the rest of the code starts without having a chat.